h2o-algos/src/main/java/hex/gram/Gram.java

```diff
@@ -731,6 +755,82 @@ public void mul(double [] x, double [] res){
     }
   }
 
+
+  /**
+   * Task to compute outer product of a matrix normalized by the number of observations (not counting rows with NAs).
+   * in R's notation g = X%*%T(X)/nobs, nobs = number of rows of X with no NA.  Copied from GramTask.
+   * @author wendycwong
+   */
+  public static class OuterGramTask extends FrameTask2<OuterGramTask> {
+    private  boolean _std = true;
+    public Gram _gram;
+    public long _nobs;
+    boolean _intercept = false;
+    int[] _catOffsets;
+    double _scale;    // 1/(number of samples)
+
+
+    public OuterGramTask(Key<Job> jobKey, DataInfo dinfo){
+      super(null,dinfo,jobKey);
+      _catOffsets = dinfo._catOffsets != null?Arrays.copyOf(dinfo._catOffsets, dinfo._catOffsets.length):null;
+      _scale = dinfo._adaptedFrame.numRows() > 0?1.0/dinfo._adaptedFrame.numRows():0.0;
+    }
+
+    /*
+    Need to do our own thing here since we need to access and multiple different rows of a chunck.
+     */
+    @Override public void map(Chunk[] chks) { // TODO: implement the sparse option.
+      chunkInit();
+
+      DataInfo.Row rowi = _dinfo.newDenseRow();
+      DataInfo.Row rowj = _dinfo.newDenseRow();
+      int rowOffset = (int) chks[0].start();   // calculate row indices for this particular chunks of data
+
+      for(int i = 0 ; i < chks[0]._len; i++) {  // each loop through here will set one element of gram matrix
+        _dinfo.extractDenseRow(chks, i, rowi);
+        if (!rowi.isBad()) {
+          ++_nobs;  // increment number of training samples used
+          int rowIOffset = i+rowOffset;
+          for (int j = 0; j <= i; j++) {
+            _dinfo.extractDenseRow(chks, j, rowj);
+
+            if ((!rowi.isBad() && rowi.weight != 0) && (!rowj.isBad() && rowj.weight != 0)) {
+              processRow(rowi, rowj, rowIOffset, j + rowOffset);
```

Comment:
This is true.  I copied operation from the Gram class since I will in the end obtain a symmetric matrix and use the same scaling of 1/n.